src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementEventService.java

```diff
@@ -156,6 +158,76 @@ private CreateUpdateRfx createUpdateRfxRequest(final ProcurementProject project,
     return new CreateUpdateRfx(OperationCode.CREATE_FROM_TEMPLATE, rfx);
   }
 
+  /**
+   * Update the name, i.e. 'short description' of a Jaggaer Rfx record.
+   * 
+   * @param projectId
+   * @param eventId CCS generated event Id, e.g. 'ocds-b5fd17-2'
+   * @param eventName the new name
+   */
+  public void updateProcurementEventName(final Integer projectId, final String eventId,
+      final String eventName, final String principal) {
+
+    Assert.hasLength(eventName, "New event name must be supplied");
+
+    // EventId will be in the format "ocds-b5fd17-1"
+    String ocdsAuthorityName;
+    String ocidPrefix;
+    Integer eventIdKey;
+
+    // Validate eventId
```

Comment:
Wasn't thrilled with this bit (on extracting the individual items from the concatenated id)- did experiment with using an @Embeddable to make all 3 of these fields into an object that could be treated as a single item for converting to/from the concatenated ID - but given one was the primary @Id - didn't seem to like it - and then thought it was maybe overkill. Happily take any suggestions to improve

Reply:
Only that perhaps matching a regex pattern then extracting the capture groups *might* be more reliable / concise.. or just more of a headache! 😃

Reply:
i tend to equate regex with headaches - but I will take a look - certainly should be more concise 😃